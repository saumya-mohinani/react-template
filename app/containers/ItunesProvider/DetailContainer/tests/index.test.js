/**
 *
 * Tests for DetailContainer
 *
 *
 */

import React from 'react';
import { renderProvider, timeout } from '@utils/testUtils';
import { DetailContainerTest as DetailContainer } from '../index';

describe('<DetailContainer /> container tests', () => {
  let submitSpy;
  jest.mock('react-router-dom', () => ({
    __esModule: true,
    useParams: jest.fn().mockReturnValue({ trackId: '1' })
  }));
  beforeEach(() => {
    submitSpy = jest.fn();
  });
  it('should render and match the snapshot', () => {
    const { baseElement } = renderProvider(<DetailContainer dispatchTrack={submitSpy} />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should call dispatchTrack on render', async () => {
    renderProvider(<DetailContainer dispatchTrack={submitSpy} />);
    await timeout(500);
    expect(submitSpy).toBeCalled();
  });

  it('should call dispatchClearTrack on unmount', async () => {
    const clearSpy = jest.fn();
    const submitSpy = jest.fn();
    const { unmount } = renderProvider(<DetailContainer dispatchClearTrack={clearSpy} dispatchTrack={submitSpy} />);
    await timeout(500);
    expect(submitSpy).toBeCalled();
    unmount();
    await timeout(500);
    expect(clearSpy).toBeCalled();
  });
});
