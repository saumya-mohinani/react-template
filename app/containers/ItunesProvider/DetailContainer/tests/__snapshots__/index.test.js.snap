// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailContainer /> container tests should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="DetailContainer__ResultContainer-luq84g-3 gwxbMM"
    />
  </div>
</body>
`;
